device-api/src/device.rs

```diff
@@ -447,6 +447,13 @@ impl CoreRange {
             CoreRange::Range((s, e)) => (*s..=*e).contains(idx),
         }
     }
+
+    pub fn has_intersection(&self, other: &Self) -> bool {
+        match (self, other) {
+            (CoreRange::All, _) | (_, CoreRange::All) => true,
+            (CoreRange::Range(a), CoreRange::Range(b)) => !(a.0 > b.1 || a.1 < b.0),
```

Comment:
```suggestion
            (CoreRange::Range(a), CoreRange::Range(b)) => !(a.1 < b.0 || b.1 < a.0 ),
```
for readability,

Reply:
Reflect at [c87341f](https://github.com/furiosa-ai/device-api/pull/101/commits/c87341f0539551194ad58ee8c76fa10425f64540)